### ISSUE TITLE
Cannot complete ./certbot-auto (pypi website misbehaving)

### ISSUE THREAD

**Issue by GotDemBallsOfSteel** [certbot/certbot#3636]
Notice: I saw this ticket #3313    
  
```
Creating virtual environment...
Installing Python packages...
Traceback (most recent call last):
  File "/tmp/tmp.GdEQLWJYU2/pipstrap.py", line 146, in <module>
    exit(main())
  File "/tmp/tmp.GdEQLWJYU2/pipstrap.py", line 130, in main
    for url, digest in PACKAGES]
  File "/tmp/tmp.GdEQLWJYU2/pipstrap.py", line 112, in hashed_download
    response = opener().open(url)
  File "/usr/lib/python2.7/urllib2.py", line 404, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 422, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 1222, in https_open
    return self.do_open(httplib.HTTPSConnection, req)
  File "/usr/lib/python2.7/urllib2.py", line 1184, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [Errno 104] Connection reset by peer>

```
  
I can "curl" the url you gave back then at that ticket.  
My server can connect outside, people connect to it (webserver)... so no idea.  
Using OpenDNS.  
  
OS: Ubuntu 14.0.4.5 LTS amd64

**Comment by bmw** [comment]
Huh. Have you tried running `certbot-auto` again and gotten the same error? I want to make sure there wasn't a hiccup in our script, PyPI, the internet, etc. that just caused a one time problem.

You said `curl https://pypi.python.org/pypi/certbot/json` worked properly, What was the output? I'd like to make sure nothing strange is going on here.

If you're still having the issue, I'd like you to play with this simple script:
```
#!/usr/bin/env python
import sys 
try:
    from urllib2 import build_opener, HTTPHandler, HTTPSHandler
except ImportError:
    from urllib.request import build_opener, HTTPHandler, HTTPSHandler
opener = build_opener(HTTPSHandler())
# Strip out HTTPHandler to prevent MITM spoof:
for handler in opener.handlers:
    if isinstance(handler, HTTPHandler):
        opener.handlers.remove(handler)
opener.open(sys.argv[1])
print "success!"
```
Put that in a file named something like `open.py` and run `python open.py <HTTPS URL here>`. For example you could run `python open.py https://certbot.eff.org` or `python open.py https://google.com`. Do you see `success!` or a traceback?

If you see `success!`, what about when running
`python open.py https://pypi.python.org/packages/source/p/pip/pip-8.0.3.tar.gz`?

**Comment by GotDemBallsOfSteel** [comment]
Yeah I give up, it works now. It seems that the pypi website is just unreliable.  
I tried it yesterday several times, and the servers are hosted in a major datacenter, so their connection should be top notch.  
  
Sorry for opening the ticket.  
I think a warning should just appear that if the script fails, try again a day later or something.  
Luckily I was able to renew my cert just one day before it expired.   
  
Thanks for your great work, certbot is amazing.
